components/infra-proxy-service/server/environments.go

```diff
@@ -143,7 +142,65 @@ func (s *Server) DeleteEnvironment(ctx context.Context, req *request.Environment
 	return &response.Environment{
 		Name: environment.Name,
 	}, nil
+}
+
+// UpdateEnvironment updates the environment attributes
+func (s *Server) UpdateEnvironment(ctx context.Context, req *request.UpdateEnvironment) (*response.Environment, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "must supply environment name")
+	}
+
+	cookbooks := req.CookbookVersions
+	if cookbooks == nil {
+		cookbooks = map[string]string{}
+	}
+
+	var defAtt interface{}
+	var ovrAtt interface{}
+
+	defIn := req.DefaultAttributes
+	if defIn == "" {
+		defIn = "{}"
+	}
```

Comment:
This is used so many times already; could we make a helper function and fix all occurrences? Then this becomes, for instance:
```
defIn = defaultIfEmpty(req.DefaultAttributes)
```